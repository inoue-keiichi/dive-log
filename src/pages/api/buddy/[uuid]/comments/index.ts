import { prisma } from "@/clients/prisma";
import { buddyCommentQuerySchema, buddyCommentSchema } from "@/schemas/buudy";
import type { NextApiRequest, NextApiResponse } from "next";

export default async function handler(
  req: NextApiRequest,
  res: NextApiResponse<{}>
) {
  if (req.method === "GET") {
    const parsedQuery = buddyCommentQuerySchema.safeParse({
      uuid: req.query.uuid,
    });
    if (!parsedQuery.success) {
      // TODO: エラーのレスポンス型作りたい
      return res.status(400).json({
        errorCode: "invalid_parameter",
        message: JSON.parse(parsedQuery.error.message),
      });
    }

    const result = await prisma.diveLogLink.findFirst({
      include: {
        diveLog: {
          include: {
            buddyComments: true,
          },
        },
      },
      where: {
        uuid: parsedQuery.data.uuid,
      },
    });

    if (!result) {
      return res.status(500).json({});
    }

    return res.status(200).json(result.diveLog.buddyComments);
  }

  return res.status(400).json({});
}
