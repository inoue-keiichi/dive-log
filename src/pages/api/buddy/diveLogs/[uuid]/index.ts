import { prisma } from "@/clients/prisma";
import { buddyCommentQuerySchema } from "@/schemas/buudy";
import { ResponseError } from "@/utils/type";
import type { NextApiRequest, NextApiResponse } from "next";

export default async function handler(
  req: NextApiRequest,
  res: NextApiResponse<ShareDiveLog | ResponseError>
) {
  if (req.method === "GET") {
    const parsedQuery = buddyCommentQuerySchema.safeParse({
      uuid: req.query.uuid,
    });
    if (!parsedQuery.success) {
      // TODO: エラーのレスポンス型作りたい
      return res.status(400).json({
        code: "invalid_parameter",
        message: parsedQuery.error.message,
      });
    }

    const result = await prisma.diveLogLink.findFirst({
      include: {
        diveLog: {
          include: {
            buddies: {
              include: {
                guest: true,
                comments: true,
              },
            },
          },
        },
      },
      where: {
        uuid: parsedQuery.data.uuid,
        expiredAt: {
          gte: new Date(),
        },
      },
    });

    if (!result) {
      return res.status(400).json({
        code: "resource_not_found",
        message:
          "有効期限が切れた、または不正なURLです。バディに再度URLを発行してもらってください",
      });
    }

    const { date, place, point, divingStartTime, divingEndTime, buddies } =
      result.diveLog;
    const buddyComments = buddies.flatMap((buddy) => {
      const { comments, guest } = buddy;
      return comments.map((comment) => {
        const { buddyId, ...others } = comment;
        return {
          name: guest!.name,
          ...others,
        };
      });
    });

    return res.status(200).json({
      date,
      place,
      point,
      divingStartTime,
      divingEndTime,
      buddyComments: [...buddyComments].sort(
        (a, b) => a.createdAt.getTime() - b.createdAt.getTime()
      ),
    });
  }

  return res.status(400).json({
    code: "invaid_http_method",
  });
}

export type ShareDiveLog = {
  date: string;
  place: string | null;
  point: string | null;
  divingStartTime: string | null;
  divingEndTime: string | null;
  buddyComments: {
    id: number;
    name: string;
    text: string;
    createdAt: Date;
  }[];
};
