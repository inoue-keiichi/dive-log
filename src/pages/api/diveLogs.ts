import { prisma } from "@/clients/prisma";
import { DiveLog } from "@/domains/diveLog";
import { PrismaClient } from "@prisma/client";
import type { NextApiRequest, NextApiResponse } from "next";

export default async function handler(
  req: NextApiRequest,
  res: NextApiResponse<{}>
) {
  if (req.method === "POST") {
    // TODO: zod
    const { userId, point, waterTemprature, transparency } = JSON.parse(
      req.body
    ) as {
      userId: string;
      point: string;
      waterTemprature: string;
      transparency: string;
    };
    await prisma.diveLog.create({
      data: {
        userId,
        point,
        waterTemprature: Number(waterTemprature),
        transparency: Number(transparency),
      },
    });

    return res.status(200).json({});
  }

  if (req.method === "GET") {
    // TODO: zod
    const diveLogs = await prisma.diveLog.findMany({
      where: { userId: req.query.userId as string },
      orderBy: { id: "desc" },
    });
    return res.status(200).json(diveLogs);
  }

  return res.status(400).json({});
}
