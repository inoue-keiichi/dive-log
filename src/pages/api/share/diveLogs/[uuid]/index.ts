import { prisma } from "@/clients/prisma";
import { diveLogLinkSchema } from "@/schemas/buudy";
import { ResponseError } from "@/utils/type";
import type { NextApiRequest, NextApiResponse } from "next";

export default async function handler(
  req: NextApiRequest,
  res: NextApiResponse<ShareDiveLog | ResponseError>
) {
  if (req.method === "GET") {
    const parsedQuery = diveLogLinkSchema.safeParse({
      uuid: req.query.uuid,
    });
    if (!parsedQuery.success) {
      // TODO: エラーのレスポンス型作りたい
      return res.status(400).json({
        code: "invalid_parameter",
        message: parsedQuery.error.message,
      });
    }

    const result = await prisma.diveLogLink.findFirst({
      include: {
        diveLog: {
          include: {
            buddies: {
              include: {
                guest: true,
                comments: true,
              },
            },
          },
        },
      },
      where: {
        uuid: parsedQuery.data.uuid,
        expiredAt: {
          gte: new Date(),
        },
      },
    });

    if (!result) {
      return res.status(400).json({
        code: "resource_not_found",
        message:
          "有効期限が切れた、または不正なURLです。バディに再度URLを発行してもらってください",
      });
    }

    const { date, place, point, divingStartTime, divingEndTime, buddies } =
      result.diveLog;

    return res.status(200).json({
      date,
      place,
      point,
      divingStartTime,
      divingEndTime,
      buddies: buddies.map((buddy) => ({
        id: buddy.id,
        name: buddy.guest!.name, // ユーザー登録機能ができるまではnullにならない
        comments: buddy.comments.map((comment) => ({
          text: comment.text,
          createdAt: comment.createdAt,
        })),
      })),
    });
  }

  return res.status(400).json({
    code: "invaid_http_method",
  });
}

export type ShareDiveLog = {
  date: string;
  place: string | null;
  point: string | null;
  divingStartTime: string | null;
  divingEndTime: string | null;
  buddies: {
    id: number;
    name: string;
    comments: {
      text: string;
      createdAt: Date;
    }[];
  }[];
};
