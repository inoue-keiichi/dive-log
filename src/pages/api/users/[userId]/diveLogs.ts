import { prisma } from "@/clients/prisma";
import { newDiveLogQuerySchema, diveLogSchema } from "@/schemas/diveLog";
import type { NextApiRequest, NextApiResponse } from "next";

export default async function handler(
  req: NextApiRequest,
  res: NextApiResponse<{}>
) {
  if (req.method === "POST") {
    const parsedQuery = newDiveLogQuerySchema.safeParse({
      userId: req.query.userId,
    });
    if (!parsedQuery.success) {
      // TODO: エラーのレスポンス型作りたい
      return res.status(400).json({
        errorCode: "invalid_parameter",
        message: JSON.parse(parsedQuery.error.message),
      });
    }

    const parsed = diveLogSchema.safeParse(JSON.parse(req.body));
    if (!parsed.success) {
      // TODO: エラーのレスポンス型作りたい
      return res.status(400).json({
        errorCode: "invalid_parameter",
        message: JSON.parse(parsed.error.message),
      });
    }

    const newDiveLog = await prisma.diveLog.create({
      data: { ...parsed.data, userId: parsedQuery.data.userId },
    });

    return res.status(201).json(newDiveLog);
  }

  if (req.method === "GET") {
    // TODO: zod
    const diveLogs = await prisma.diveLog.findMany({
      where: { userId: req.query.userId as string },
      orderBy: { id: "desc" },
    });
    return res.status(200).json(diveLogs);
  }

  return res.status(400).json({});
}
