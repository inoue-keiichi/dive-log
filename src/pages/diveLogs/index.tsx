import DiveLogLinkDialog from "@/components/templates/DiveLogLinkDialog";
import DiveLogList from "@/components/templates/DiveLogList";
import { DiveLog } from "@/schemas/diveLog";
import styles from "@/styles/Home.module.css";
import { createServerSupabaseClient } from "@supabase/auth-helpers-nextjs";
import { useUser } from "@supabase/auth-helpers-react";
import { GetServerSidePropsContext } from "next";
import Head from "next/head";
import { useRouter } from "next/router";
import { useState } from "react";
import { LineIcon, LineShareButton } from "react-share";

type Props = {
  diveLogs: (DiveLog & { id: number })[];
};

export default function DivingLogs(props: Props) {
  const { diveLogs } = props;

  const [openShareDialog, setOpenShareDialog] = useState<boolean>(false);
  const [link, setLink] = useState<string>("");

  const router = useRouter();

  const user = useUser();

  return (
    <>
      <Head>
        <title>Create Next App</title>
        <meta name="description" content="Generated by create next app" />
        <meta name="viewport" content="width=device-width, initial-scale=1" />
        <link rel="icon" href="/favicon.ico" />
      </Head>
      <main className={styles.main}>
        <DiveLogLinkDialog
          open={openShareDialog}
          link={link}
          onClose={setOpenShareDialog}
        >
          <LineShareButton
            title={"ログブックにコメントを書いてね！"}
            url={link}
          >
            <LineIcon size={32} borderRadius={10} />
          </LineShareButton>
        </DiveLogLinkDialog>
        <DiveLogList
          data-test-id={"dive-log-card-list"}
          diveLogs={diveLogs}
          onAddNew={() => router.push("/diveLogs/new")}
          onEdit={(id: number) =>
            router.push({ pathname: "/diveLogs/[id]", query: { id } })
          }
          onShare={async (id: number) => {
            if (!user) {
              await router.replace("/");
              return;
            }

            const res = await fetch(
              `${process.env.NEXT_PUBLIC_HOST}/api/users/${user.id}/diveLogs/${id}/buddy/shareLink/issue`,
              { method: "POST" }
            );
            if (!res.ok) {
              // TODO: エラー処理かく.link取得できない場合、リカバリー策あるか？
              throw new Error();
            }
            const result = (await res.json()) as { link: string } | null;
            if (!result) {
              // TODO: エラー処理かく.link取得できない場合、リカバリー策あるか？
              throw new Error();
            }
            setLink(result.link);
            setOpenShareDialog(true);
          }}
        />
      </main>
    </>
  );
}

export async function getServerSideProps(context: GetServerSidePropsContext) {
  const supabaseServerClient = createServerSupabaseClient(context);
  const {
    data: { user },
  } = await supabaseServerClient.auth.getUser();

  if (!user) {
    return {
      redirect: {
        destination: "/",
        permanent: false,
      },
    };
  }

  const res = await fetch(
    `${process.env.NEXT_PUBLIC_HOST}/api/users/${user.id}/diveLogs`
  );
  const diveLogs = (await res.json()) as DiveLog[];
  return { props: { diveLogs } };
}
