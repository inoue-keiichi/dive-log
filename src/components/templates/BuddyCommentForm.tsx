import { ShareDiveLog } from "@/pages/api/share/diveLogs/[uuid]";
import { BuddyComment, buddyCommentSchema } from "@/schemas/buudy";
import { DiveLog } from "@/schemas/diveLog";
import { ResponseError } from "@/utils/type";
import { zodResolver } from "@hookform/resolvers/zod";
import {
  Button,
  Chip,
  Divider,
  FormControl,
  FormHelperText,
  InputLabel,
  List,
  ListItem,
  ListItemText,
  OutlinedInput,
  Stack,
  TextField,
  Typography,
} from "@mui/material";
import dayjs from "dayjs";
import { FC } from "react";
import { FieldErrors, useForm } from "react-hook-form";

const hasInvalidField = (errors: FieldErrors<DiveLog>) => {
  return Object.values(errors).filter((v) => v.message).length > 0;
};

type Props = {
  commenter: string;
  diveLog: ShareDiveLog;
  onSubmit: (buddyComment: BuddyComment) => void;
  error?: ResponseError;
};

const BuddyCommentForm: FC<Props> = (props) => {
  const { commenter, onSubmit, diveLog, error } = props;

  const buddyComments = diveLog.buddies.flatMap((buddy) =>
    buddy.comments.map((comment) => ({
      name: buddy.name,
      id: comment.id,
      text: comment.text,
      createdAt: comment.createdAt,
    }))
  );

  const {
    register,
    handleSubmit,
    formState: { errors },
  } = useForm<BuddyComment>({
    resolver: zodResolver(buddyCommentSchema),
  });

  const divingBetweenTimeStr =
    diveLog.divingStartTime && diveLog.divingEndTime
      ? `${diveLog.divingStartTime || ""}~${diveLog.divingEndTime || ""}`
      : "";

  return (
    <Stack
      sx={{ width: "100%", backgroundColor: "white", borderRadius: 2 }}
      spacing={2}
    >
      <Stack
        spacing={2}
        sx={{
          padding: "20px 40px 20px 40px",
          borderRadius: 2,
        }}
        component="form"
        onSubmit={handleSubmit((buddyComment) => {
          onSubmit(buddyComment);
        })}
      >
        <Typography variant="h5">{`${diveLog.date} ${divingBetweenTimeStr}`}</Typography>
        <Stack direction="row" spacing={1}>
          {diveLog.place && <Chip label={diveLog.place} color="primary" />}
          {diveLog.point && <Chip label={diveLog.point} color="primary" />}
        </Stack>
        <FormControl>
          <TextField
            id="name"
            label="名前"
            type="text"
            defaultValue={commenter}
            InputProps={{
              readOnly: true,
            }}
          />
        </FormControl>
        <FormControl>
          <InputLabel htmlFor="text">コメント</InputLabel>
          <OutlinedInput
            id="text"
            label="コメント"
            type="text"
            multiline
            minRows={3}
            error={!!errors.text?.message}
            {...register("text")}
          />
          <FormHelperText error={!!errors.text?.message}>
            {errors.text?.message ?? ""}
          </FormHelperText>
          <FormHelperText error={!!error}>
            {error?.message ?? ""}
          </FormHelperText>
        </FormControl>
        <Button
          sx={{ width: "30%" }}
          disabled={!!error || hasInvalidField(errors)}
          variant="contained"
          type="submit"
        >
          送信
        </Button>
      </Stack>
      {buddyComments.length > 0 && (
        <List sx={{ bgcolor: "background.paper", borderRadius: 2 }}>
          {buddyComments.map((comment) => (
            <div key={`coment_list_item_${comment.id}`}>
              <Divider />
              <ListItem alignItems="flex-start">
                <ListItemText
                  primary={comment.text}
                  secondary={
                    comment.name +
                    " — " +
                    dayjs(comment.createdAt).format("YYYY/M/D H:mm")
                  }
                />
              </ListItem>
            </div>
          ))}
        </List>
      )}
    </Stack>
  );
};

export default BuddyCommentForm;
