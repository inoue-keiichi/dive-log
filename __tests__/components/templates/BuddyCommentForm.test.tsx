import BuddyCommentForm from "@/components/templates/BuddyCommentForm";
import { ShareDiveLog } from "@/pages/api/share/diveLogs/[uuid]";
import { fireEvent, render, screen, waitFor } from "@testing-library/react";
import { fill } from "../__utils__/common";

const submit = jest.fn();
afterEach(() => {
  submit.mockClear();
});

describe("BuddyCommentForm", () => {
  // TODO: clipboardのテスト書く
  test("succeeded in submitting a comment", async () => {
    const diveLog: ShareDiveLog = {
      date: "2023-08-23",
      place: null,
      point: null,
      divingStartTime: null,
      divingEndTime: null,
      buddies: [
        {
          id: 1,
          name: "JAI",
          comments: [],
        },
      ],
    };

    render(
      <BuddyCommentForm diveLog={diveLog} onSubmit={submit} commenter={"JAI"} />
    );
    fill(screen.getByLabelText("コメント"), "楽しいダイビングだった。");
    const submitButton = screen.getByText("送信");
    fireEvent.click(submitButton);
    // react-hook-form によって submit が呼び出されるまで待機
    await waitFor(() => expect(submit).toHaveBeenCalledTimes(1));
    expect(submit.mock.calls[0][0]).toStrictEqual({
      text: "楽しいダイビングだった。",
    });
  });

  test("cannot submit without a comment", async () => {
    const diveLog: ShareDiveLog = {
      date: "2023-08-23",
      place: null,
      point: null,
      divingStartTime: null,
      divingEndTime: null,
      buddies: [
        {
          id: 1,
          name: "JAI",
          comments: [],
        },
      ],
    };

    render(
      <BuddyCommentForm diveLog={diveLog} onSubmit={submit} commenter={"JAI"} />
    );
    const submitButton = screen.getByText("送信");
    fireEvent.click(submitButton);
    // react-hook-form によって submit が呼び出されるまで待機
    await waitFor(() => expect(submit).toHaveBeenCalledTimes(0));
  });
});

describe("BuddyCommentForm snapshot", () => {
  test("Comment Form with other bubby comments and the diveLog", async () => {
    const diveLog: ShareDiveLog = {
      date: "2023-08-23",
      place: "大瀬",
      point: "湾内",
      divingStartTime: "09:30",
      divingEndTime: "10:00",
      buddies: [
        {
          id: 1,
          name: "鈴木",
          comments: [
            {
              id: 1,
              text: "今日の海は味噌汁だった。",
              createdAt: new Date("2023-08-01"),
            },
          ],
        },
        {
          id: 2,
          name: "原田",
          comments: [
            {
              id: 2,
              text: "海底はまだ冷たくて早く陸に帰りたかった...",
              createdAt: new Date("2023-08-01"),
            },
          ],
        },
      ],
    };

    const tree = render(
      <BuddyCommentForm diveLog={diveLog} onSubmit={submit} commenter={"JAI"} />
    );
    expect(tree).toMatchSnapshot();
  });
});
