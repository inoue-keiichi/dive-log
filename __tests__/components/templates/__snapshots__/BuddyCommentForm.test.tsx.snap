// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BuddyCommentForm snapshot Comment Form with other bubby comments and the diveLog 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiStack-root css-89v2n4-MuiStack-root"
      >
        <form
          class="MuiStack-root css-10w7mp6-MuiStack-root"
        >
          <h5
            class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
          >
            2023-08-23 09:30~10:00
          </h5>
          <div
            class="MuiStack-root css-w4z10b-MuiStack-root"
          >
            <div
              class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorPrimary MuiChip-filledPrimary css-1sz23ua-MuiChip-root"
            >
              <span
                class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
              >
                大瀬
              </span>
            </div>
            <div
              class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorPrimary MuiChip-filledPrimary css-1sz23ua-MuiChip-root"
            >
              <span
                class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
              >
                湾内
              </span>
            </div>
          </div>
          <div
            class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
          >
            <div
              class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="true"
                for="name"
                id="name-label"
              >
                名前
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl Mui-readOnly MuiInputBase-readOnly css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input Mui-readOnly MuiInputBase-readOnly css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  id="name"
                  readonly=""
                  type="text"
                  value="JAI"
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-14lo706"
                  >
                    <span>
                      名前
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for="text"
            >
              コメント
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-multiline css-dpjnhs-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <textarea
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
                id="text"
                name="text"
                style="height: 0px; overflow: hidden;"
              />
              <textarea
                aria-hidden="true"
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
                readonly=""
                style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); padding: 0px; width: 100%;"
                tabindex="-1"
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-yjsfm1"
                >
                  <span>
                    コメント
                  </span>
                </legend>
              </fieldset>
            </div>
            <p
              class="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained css-1wc848c-MuiFormHelperText-root"
            />
            <p
              class="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained css-1wc848c-MuiFormHelperText-root"
            />
          </div>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1u4rjth-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="submit"
          >
            送信
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </form>
        <ul
          class="MuiList-root MuiList-padding css-14qn28a-MuiList-root"
        >
          <div>
            <hr
              class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
            />
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding MuiListItem-alignItemsFlexStart css-hhvo2v-MuiListItem-root"
            >
              <div
                class="MuiListItemText-root MuiListItemText-multiline css-konndc-MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                >
                  今日の海は味噌汁だった。
                </span>
                <p
                  class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-83ijpv-MuiTypography-root"
                >
                  鈴木 — 2023/8/1 9:00
                </p>
              </div>
            </li>
          </div>
          <div>
            <hr
              class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
            />
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding MuiListItem-alignItemsFlexStart css-hhvo2v-MuiListItem-root"
            >
              <div
                class="MuiListItemText-root MuiListItemText-multiline css-konndc-MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                >
                  海底はまだ冷たくて早く陸に帰りたかった...
                </span>
                <p
                  class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-83ijpv-MuiTypography-root"
                >
                  原田 — 2023/8/1 9:00
                </p>
              </div>
            </li>
          </div>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiStack-root css-89v2n4-MuiStack-root"
    >
      <form
        class="MuiStack-root css-10w7mp6-MuiStack-root"
      >
        <h5
          class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
        >
          2023-08-23 09:30~10:00
        </h5>
        <div
          class="MuiStack-root css-w4z10b-MuiStack-root"
        >
          <div
            class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorPrimary MuiChip-filledPrimary css-1sz23ua-MuiChip-root"
          >
            <span
              class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
            >
              大瀬
            </span>
          </div>
          <div
            class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorPrimary MuiChip-filledPrimary css-1sz23ua-MuiChip-root"
          >
            <span
              class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
            >
              湾内
            </span>
          </div>
        </div>
        <div
          class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
        >
          <div
            class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="true"
              for="name"
              id="name-label"
            >
              名前
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl Mui-readOnly MuiInputBase-readOnly css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input Mui-readOnly MuiInputBase-readOnly css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                id="name"
                readonly=""
                type="text"
                value="JAI"
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-14lo706"
                >
                  <span>
                    名前
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for="text"
          >
            コメント
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-multiline css-dpjnhs-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <textarea
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
              id="text"
              name="text"
              style="height: 0px; overflow: hidden;"
            />
            <textarea
              aria-hidden="true"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
              readonly=""
              style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); padding: 0px; width: 100%;"
              tabindex="-1"
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-yjsfm1"
              >
                <span>
                  コメント
                </span>
              </legend>
            </fieldset>
          </div>
          <p
            class="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained css-1wc848c-MuiFormHelperText-root"
          />
          <p
            class="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained css-1wc848c-MuiFormHelperText-root"
          />
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1u4rjth-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="submit"
        >
          送信
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </form>
      <ul
        class="MuiList-root MuiList-padding css-14qn28a-MuiList-root"
      >
        <div>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
          />
          <li
            class="MuiListItem-root MuiListItem-gutters MuiListItem-padding MuiListItem-alignItemsFlexStart css-hhvo2v-MuiListItem-root"
          >
            <div
              class="MuiListItemText-root MuiListItemText-multiline css-konndc-MuiListItemText-root"
            >
              <span
                class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
              >
                今日の海は味噌汁だった。
              </span>
              <p
                class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-83ijpv-MuiTypography-root"
              >
                鈴木 — 2023/8/1 9:00
              </p>
            </div>
          </li>
        </div>
        <div>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
          />
          <li
            class="MuiListItem-root MuiListItem-gutters MuiListItem-padding MuiListItem-alignItemsFlexStart css-hhvo2v-MuiListItem-root"
          >
            <div
              class="MuiListItemText-root MuiListItemText-multiline css-konndc-MuiListItemText-root"
            >
              <span
                class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
              >
                海底はまだ冷たくて早く陸に帰りたかった...
              </span>
              <p
                class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-83ijpv-MuiTypography-root"
              >
                原田 — 2023/8/1 9:00
              </p>
            </div>
          </li>
        </div>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
